### PR TITLE
Feature/date range filter

### PR DESCRIPTION
**Pull request by rlskoeser**
Publication date range filter for #30

### CODE REVIEW

File: ppa/archive/forms.py [hunk 1]
```diff
@@ -53,6 +114,45 @@ def set_choices_from_facets(self, facets):
...
+    def pub_date_minmax(self):
+        '''Get minimum and maximum values for
+        :class:`~ppa.archive.models.DigitizedWork` publication dates
+        in the database.  Used to set placeholder values for the form
+        input and to generate the Solr facet range query.
+        Value is cached to avoid repeatedly calculating it.
+
+        :returns: tuple of min, max
+        '''
+        # maxmin = cache.get(self.PUBDATE_CACHE_KEY)
```
Comment (bwhicks) [inline ppa/archive/forms.py]: Am I looking at this wrongly or would this line being commented out result in `pub_date_minmax` never pulling in the value from cache?

Comment (rlskoeser) [reply ppa/archive/forms.py]: @bwhicks whoa, thanks for catching that.  I must have commented it out when I was testing some error handling.

File: ppa/archive/views.py [hunk 2]
```diff
@@ -81,7 +87,47 @@ def get_queryset(self, **kwargs):
...
+
+            # get minimum and maximum pub date values from the db
+            pubmin, pubmax = self.form.pub_date_minmax()
+
+            start = int(start) if start else pubmin
+            end = int(end) if end else pubmax
+
+            # fallback logic for when start and end are not set
+            # (should only happen when no content is in the database)
+            if not start or end:
```
Comment (thatbudakguy) [inline ppa/archive/views.py]: how can we get here after lines 105-106? if start/end are `None` won't they be set above?

Comment (thatbudakguy) [reply ppa/archive/views.py]: is this supposed to be `if not pubmin or pubmax`, etc?

Comment (rlskoeser) [reply ppa/archive/views.py]: Good point.  Just simplified to 
```
start = int(start) if start else pubmin or 0
end = int(end) if end else pubmax or 1922
```

**Pull request merged**